File: ios/Plugin/CapacitorGoogleAnalyticsPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>
#import <GoogleAnalytics/GAI.h>
#import <GoogleAnalytics/GAIDictionaryBuilder.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(CapacitorGoogleAnalyticsPlugin, "CapacitorGoogleAnalytics",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(create, CAPPluginReturnPromise);
)

File: ios/Plugin/CapacitorGoogleAnalyticsPlugin.swift
import Foundation
import Capacitor
import GoogleAnalytics

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */

let resultKey = "result"

@objc(CapacitorGoogleAnalyticsPlugin)
public class CapacitorGoogleAnalyticsPlugin: CAPPlugin {
    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve(["value": value])
    }
}
